### PR TITLE
feat: support get workflow task execution status

### LINKED ISSUE #6350: Need a API to get workflow tasks already run
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

I'm making a webapp as a front-end and using dify as backend. When the workflow request sent , I don't want users to wait the result. I can use the streaming mode to start the workflow run and get the task_id. I want users back to get the result after maybe 5 or 10 mins. I can't found the api to get the task results by `task_id` or `workflow_run_id`.

The API maybe should look like this :
```
//get tasks already run in page 3 and 10 rows
curl -X GET 'http://localhost/v1/workflows/tasks' \
-H 'Authorization: Bearer {api_key}' \
-H 'Content-Type: application/json' \
--data-raw '{"user": "abc-123","offset":20,"limit":10}'

//get task by taskid
curl -X GET 'http://localhost/v1/workflows/tasks/:task_id' \
-H 'Authorization: Bearer {api_key}' \
-H 'Content-Type: application/json' \
--data-raw '{"user": "abc-123"}'
```

### 2. Additional context or comments

_No response_

### 3. Can you help us with this feature?

- [ ] I am interested in contributing to this feature.

### PR DESCRIPTION
**Pull request by hjlarry**
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [x] Please open an issue before creating a PR or link to an existing issue
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description

Fixes  https://github.com/langgenius/dify/issues/6350

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [ ] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# Testing Instructions

image

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: api/controllers/service_api/app/workflow.py [hunk 1]
```diff
@@ -71,6 +73,35 @@ def post(self, app_model: App, end_user: EndUser):
             raise InternalServerError()
 
 
+class WorkflowTaskDetailApi(Resource):
```
Comment (crazywoola) [inline api/controllers/service_api/app/workflow.py]:
```suggestion
class WorkflowRunDetailApi(Resource):
```

File: api/controllers/service_api/app/workflow.py [hunk 2]
```diff
@@ -89,4 +120,5 @@ def post(self, app_model: App, end_user: EndUser, task_id: str):
 
 
 api.add_resource(WorkflowRunApi, '/workflows/run')
+api.add_resource(WorkflowTaskDetailApi, '/workflows/tasks/<string:task_id>')
```
Comment (crazywoola) [inline api/controllers/service_api/app/workflow.py]:
```suggestion
api.add_resource(WorkflowRunDetailApi, '/workflows/run/<string:workflow_run_id>')
```

File: api/controllers/service_api/app/workflow.py [hunk 3]
```diff
@@ -71,6 +73,35 @@ def post(self, app_model: App, end_user: EndUser):
...
+        'total_steps': fields.Integer,
+        'total_tokens': fields.Integer,
+        'created_at': fields.DateTime,
+        'finished_at': fields.DateTime,
+        'elapsed_time': fields.Float,
+    }
+
+    @validate_app_token
+    @marshal_with(workflow_run_fields)
+    def get(self, app_model: App, task_id: str):
```
Comment (crazywoola) [inline api/controllers/service_api/app/workflow.py]:
```suggestion
    def get(self, app_model: App, workflow_run_id: str):
```

File: api/controllers/service_api/app/workflow.py [hunk 4]
```diff
@@ -71,6 +73,35 @@ def post(self, app_model: App, end_user: EndUser):
...
+        """
+        Get a workflow task running detail
+        """
+        app_mode = AppMode.value_of(app_model.mode)
+        if app_mode != AppMode.WORKFLOW:
+            raise NotWorkflowAppError()
+
+        workflow_run = db.session.query(WorkflowRun).filter(WorkflowRun.id == task_id).first()
```
Comment (crazywoola) [inline api/controllers/service_api/app/workflow.py]:
```suggestion
        workflow_run = db.session.query(WorkflowRun).filter(WorkflowRun.id == workflow_run_id).first()
```

**Review by crazywoola — CHANGES_REQUESTED** [review]
Please change the docs as well.

**Comment by hjlarry** [comment]
now I combin them into one class, `post` is for run task, `get` is for get its running details, please review again @crazywoola

**Review by crazywoola — APPROVED** [review]

**Pull request merged**